app/src/main/java/com/example/intandem/fragments/PostsFragment.java

```diff
@@ -207,18 +354,82 @@ public void done(List<Post> posts, ParseException e) {
                             }
                         }
                         allPosts.addAll(posts);
-                        adapter.notifyDataSetChanged();
+                        if (maxDistance == -1) {
+                            adapter.notifyDataSetChanged();
+                            return;
+                        }
+                        // FILTERING BY DISTANCE
+                        filteredDistancePosts.clear();
+
+                        Retrofit retrofit = new Retrofit.Builder()
+                                .baseUrl(BASE_URL)
+                                .addConverterFactory(GsonConverterFactory.create())
+                                .build();
+                        StringBuilder destinations = new StringBuilder();
+                        for (int i = 0; i < allPosts.size() - 1; i++) {
+                            CustomPlace customPlace = allPosts.get(i).getCustomPlace();
+                            try {
+                                customPlace.fetchIfNeeded();
+                                destinations.append("place_id:").append(customPlace.getGPlaceId()).append("|");
+                            } catch (ParseException ex) {
+                                Log.e(TAG, ex.toString());
+                                ex.printStackTrace();
+                            }
+                        }
+                        CustomPlace customPlace = allPosts.get(allPosts.size() - 1).getCustomPlace();
+                        try {
+                            customPlace.fetchIfNeeded();
+                            destinations.append("place_id:").append(customPlace.getGPlaceId());
+                        } catch (ParseException ex) {
+                            Log.e(TAG, ex.toString());
+                            ex.printStackTrace();
+                        }
+
+                        GoogleMapsService googleMapsService = retrofit.create(GoogleMapsService.class);
```

Comment:
Looks like we're creating a `GoogleMapsService` each time we make a call?  It would be more efficient to store this in a global singleton somewhere.